### PR TITLE
Jormungandr: Fix Close SOAP session

### PR DESCRIPTION
**Pull request by mbounabi**
https://jira.kisio.org/browse/NAVITIAII-2377

Performance tests done on coverage 'nord' for Atos Lille:
1000 requests (/v1/coverage/nord/poi_types/poi_type%3Aamenity%3Abicycle_rental/pois)

previous code : 71.570 seconds
new code: 71.864 seconds

### CODE REVIEW

File: source/jormungandr/jormungandr/parking_space_availability/bss/atos.py [hunk 1]
```diff
@@ -76,11 +77,18 @@ def get_informations(self, poi):
...
+    @contextmanager
+    def _get_client(self):
+        try:
+            transport = zeep.Transport(timeout=self.timeout, operation_timeout=self.timeout)
+            client = zeep.Client(self.WS_URL, transport=transport)
+            yield client
+        finally:
+            transport.session.close()
 
     def _get_client(self):
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/parking_space_availability/bss/atos.py]: You might want to delete the old implementation of `_get_client`

Comment (kinnou02) [reply source/jormungandr/jormungandr/parking_space_availability/bss/atos.py]: there is no impact to recreate the `client` at each call?

**Review by kinnou02 — APPROVED** [review]

**Review by kinnou02 — CHANGES_REQUESTED** [review]
miss click

File: source/jormungandr/jormungandr/parking_space_availability/bss/atos.py (hunk 1 above)
Comment (mbounabi) [reply source/jormungandr/jormungandr/parking_space_availability/bss/atos.py]: see first comment : no impact on performance

**Comment by mbounabi** [comment]
@kinnou02, 'miss click' can be dismissed?

**Review by kinnou02 — APPROVED** [review]

File: source/jormungandr/jormungandr/parking_space_availability/bss/atos.py [hunk 2]
```diff
@@ -76,17 +77,18 @@ def get_informations(self, poi):
...
-        client = self._get_client()
-        if not client:
-            return {}
-        all_stands = client.service.getSummaryInformationTerminals(self.id_ao)
-        return {stands.libelle: Stands(stands.nbPlacesDispo, stands.nbVelosDispo) for stands in all_stands}
+        with self._get_client() as client:
+            all_stands = client.service.getSummaryInformationTerminals(self.id_ao)
+            return {stands.libelle: Stands(stands.nbPlacesDispo, stands.nbVelosDispo) for stands in all_stands}
 
+    @contextmanager
```
Comment (xlqian) [inline source/jormungandr/jormungandr/parking_space_availability/bss/atos.py:84]: :+1:

**Review by xlqian — APPROVED** [review]

**Pull request merged**